packages/ui-components/src/map-legend/Bivariate.tsx

```diff
@@ -1,14 +1,177 @@
 import React, { memo } from 'react'
-import { LegendLayer } from './MapLegend'
+import cx from 'classnames'
+import { LegendLayerBivariate } from './MapLegend'
+import styles from './Bivariate.module.css'
+import BivariateArrows from './Bivariate-arrows'
 
 type BivariateLegendProps = {
-  layer: LegendLayer
+  layer: LegendLayerBivariate
   className?: string
 }
 
+/**
+ * ---------------------------------------------------------
+ * The following code is copy pasted from the MGL fork
+ * TODO find a way to centralize
+ */
+
+// Given breaks [[0, 10, 20, 30], [-15, -5, 0, 5, 15]]:
+//
+//                                    |   |   |   |   |
+//  if first dataset selected     [   0, 10, 20, 30  ]
+//    index returned is:            0 | 1 | 2 | 3 | 4 |
+//                                    |   |   |   |   |
+//  if 2nd dataset selected       [ -15, -5,  0,  5, 15]
+//    index returned is:            0 | 1 | 2 | 3 | 4 | 5
+//                                    |   |   |   |   |
+// Note: 0 is a special value, feature is entirely omitted
+//                                            |
+//                                       undefined
+const getBucketIndex = (breaks: number[], value: number) => {
+  let currentBucketIndex
+  for (let bucketIndex = 0; bucketIndex < breaks.length + 1; bucketIndex++) {
+    const stopValue =
+      breaks[bucketIndex] !== undefined ? breaks[bucketIndex] : Number.POSITIVE_INFINITY
```

Comment:
Did you know the [nullish coalescing operator](https://mariusschulz.com/blog/nullish-coalescing-the-operator-in-typescript) ? Do you think this is easier to read ?
```suggestion
    const stopValue = breaks?.[bucketIndex] ?? Number.POSITIVE_INFINITY
```